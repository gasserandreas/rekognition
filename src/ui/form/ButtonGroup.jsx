import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

// button groups
const StyledButtonGroup = styled.div`
  display: flex;
  justify-content: flex-end;
  padding-top: 1rem;
`;

/** Simple component to render button consistently in every form.
 * Every children (Button) gets moved to right.
 */
const ButtonGroup = (props) => (
  <StyledButtonGroup>
    {props.children}
  </StyledButtonGroup>
);

ButtonGroup.propTypes = {
  /** Specify children (Button comp) */
  children: PropTypes.node.isRequired,
};

export default ButtonGroup;
