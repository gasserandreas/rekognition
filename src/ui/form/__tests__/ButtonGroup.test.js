/* global React, shallow, mount, render, renderer, testSnapshot */

import ButtonGroup from '../ButtonGroup';

it('ButtonGroup should render correctly', () => {
  testSnapshot(<ButtonGroup><button>Click me</button></ButtonGroup>);
});

