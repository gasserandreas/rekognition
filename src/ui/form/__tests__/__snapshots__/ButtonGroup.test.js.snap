// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should render correctly 1`] = `
<div
  className="sc-bdVaJa eCCpvZ"
>
  <button>
    Click me
  </button>
</div>
`;
